src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,222 @@
+    """
+    Class which provides client APIs to the DDM service.
+
+    Usage example:
+        api = 'copy'
+        ddmReq = DDMReqTemplate(api)
+        ddm = DDM()
+        ddm.makeRequest(ddmReq, api)
+    """
+
+    def __init__(self, url=None, logger=None, configDict=None):
+        """
+        configDict:
+            Dictionary with parameters that are passed to the super class.
+        """
+        url = url or "https://dynamo.mit.edu"
+        configDict = configDict or {}
+        configDict.setdefault('endpoint', url)
+        configDict.setdefault('cacheduration', 1)  # in hours
+        configDict.setdefault('accept_type', 'application/json')
+        configDict.setdefault('content_type', 'application/json')
+        configDict['logger'] = logger if logger else logging.getLogger()
+        super(DDM, self).__init__(configDict)
+        self['logger'].debug("Initializing DDM with url: %s", self['endpoint'])
+
+    def _getResult(self, ddmReq, apiUrl, callname=""):
+
+        cachedApi = "%s.json" % callname
+
+        # striping the None values from the request before sending it to dynamo
+        ddmReq.strip()
+
+        # The Request class expects this to be a dict, otherwise it tries to
+        # encode the inputdata once again and inserts parasite information and
+        # breaks the call to dynamo.
+        ddmReq = dict(ddmReq)
+        data = self.refreshCache(cachedApi,
+                                 apiUrl,
+                                 inputdata=ddmReq,
+                                 verb='POST')
+        results = data.read()
+        data.close()
+
+        results = json.loads(results)
+        return results
+
+    def makeRequest(self, ddmReq, api=None):
+        """
+        _makeRequest_
+
+        Makes a SINGLE request to DDM
+
+        :ddmReq: Request template an instance of DDMReqTemplate class.
+        :api:    The DDM api to be queried.
+        :return: Dictionary containing the result from DDM
+        """
+
+        # try to figure out the api from the ddmReq template
+        if not api:
+            api = ddmReq.api
+
+        apiUrl = "registry/request/{}".format(api)
+        callname = "DDM-{}".format(api)
+        # TODO: here to send to the logger the rquest parameters
+        return self._getResult(ddmReq, apiUrl, callname)
+
+    def makeAggrRequests(self, ddmReqList, aggrKey=None):
+        """
+        _makeAggrRequest_
+
+        Makes several Aggregated requests to DDM. Takes the list of objects,
+        instances of ddmReqTemplate and aggregates it based on a selected key.
+        Uses the ddmReqTemplate internal Equivalence method for comparison with
+        the aggregation key excluded. The final result should be a list of unique
+        ddmRequests each one of them containing all equal ddmRequests from the
+        initial list with the values of the aggregation key accumulated in a list
+        under the same key. Creates a request per ddmReqTemplate from the list
```

Comment:
I'm not sure I understand these aggregated requests where one DDM call is made for each template?
For instance, is it meant to add many datasets under the same `copy` call? If so, would those many datasets be injected in a single HTTP call? How about the response, do we get a single request id?

Reply:
Yes @amaltaro , the Answer to all those question is Yes. And we get an object (json) with the status of all requested transfers in separate fields. 
It is a single call per aggregates DDM request. I will try to make the doc string more clear.

Reply:
Just to be more precise about what we receive from DDM, here to the current message, I attach one such aggregated request with the relative response from the server.
[ddm.log](https://github.com/dmwm/WMCore/files/4466951/ddm.log)

Reply:
The docstring was updated too.